src/Controller/AppDossierController.php

```diff
@@ -516,23 +516,31 @@ public function detailDocumentenAction(Request $request, Dossier $dossier)
      * @ParamConverter("dossier", options={"id"="dossierId"})
      * @ParamConverter("document", options={"id"="documentId"})
      */
-    public function detailDocumentAction(Request $request, Dossier $dossier, Document $document, AzureStorage $azureStorage)
+    public function detailDocumentAction(
+        Request                $request,
+        Dossier                $dossier,
+        Document               $document,
+        FileStorageSelector    $fileStorageSelector
+    ): Response
     {
         $this->checkDocumentAccess($dossier, $document);
+        $filesystem = $fileStorageSelector->getFileStorageForDossier();
+        $path = 'dossier-' . $dossier->getId() . '/' . $document->getBestandsnaam();
 
-        $blobApiResponse = $this->getDocumentBlobApiResponse($document, $azureStorage);
+        $file = $filesystem->readStream($path);
 
-        $headers = $blobApiResponse->getHeaders();
+        return new StreamedResponse(
+            function () use ($file, $filesystem, $path) {
+                $outputStream = fopen('php://output', 'wb');
 
-        return new Response(
-            $blobApiResponse->getContent()
-            ,
+                stream_copy_to_stream($file, $outputStream);
+            },
             Response::HTTP_OK,
             [
                 'Content-Transfer-Encoding', 'binary',
-                'Content-Type' => $headers['content-type'][0],
+                'Content-Type' => $filesystem->getMimetype($path),
                 'Content-Disposition' => 'inline',
```

Comment:
https://symfony.com/doc/current/components/http_foundation.html#serving-files

In de symfony docs staat een aantal interessante dingen voor het serveren van files. Bijvoorbeeld een helper voor Content-Disposition.

Een ander punt is dat ze zeggen dat als je een static file serveert, dat je dan gewoon een `BinaryFileResponse` kunt returnen.

Kun je bovenstaande docs even doornemen en checken of we er iets aan hebben? Kan de code leesbaarder maken.

Reply:
Ik heb hem op deze manier geschreven omdat Flysystem er tussen zit, waardoor ik niet altijd een fysieke file heb maar ook een file kan hebben welke remote is.

Daarnaast zorgen we met deze insteek dat we een proxy zijn voor het bestand en deze niet volledig hoeven in te lezen in het geheugen. zie de eerste paaragraaf van https://symfonycasts.com/screencast/symfony-uploads/file-streaming#script